booking.yaml

```diff
@@ -0,0 +1,516 @@
+      description: The life-cycle state of the booking (from NEW to FINISHED)
+      type: string
+      enum:
+        - NEW
+        - BOOKED
+        - CANCELLED
+        - PAID
+        - UPDATE_REQUESTED
+        - UPDATED
+        - STARTED
+        - FINISHED
+    token:
+      description: >-
+        The validity token (such as booking ID, travel ticket etc.) that MaaS
+        clients will display to validate the trip when starting the leg.
+      properties:
+        validityDuration:
+          description: >-
+            The rules that MaaS will interpret to schedule, re-validate or
+            cancel the booking.
+          type: object
+          properties:
+            from:
+              description: The starting time from which the ticket is valid
+              $ref: '#/components/schemas/time'
+            to:
+              description: The finishing time the ticket is valid for
+              $ref: '#/components/schemas/time'
+        meta:
+          description: >-
+            Arbitrary metadata the TSP may pass along the ticket to the client
+            (e.g. a booking code, base64 encoded binary)
+          type: object
+    customer:
+      type: object
+      required:
+        - id
+        - firstName
+        - lastName
+      properties:
+        id:
+          description: The identifier MaaS uses to identify the customer
+          type: string
+        firstName:
+          description: First name of the customer (e.g. John)
+          type: string
+        lastName:
+          description: Last name of the customer (e.g. Doe)
+          type: string
+        phone:
+          description: Phone number that the customer may be reached from
+          type: string
+    options:
+      type: object
+      description: Containing an array of available options matching the query
+      properties:
+        leg:
+          $ref: '#/components/schemas/options_leg'
+        meta:
+          $ref: '#/components/schemas/options_meta'
+    options_leg:
+      type: object
+      properties:
+        startTime:
+          $ref: '#/components/schemas/time'
+        endTime:
+          $ref: '#/components/schemas/time'
+        from:
+          $ref: '#/components/schemas/options_coordinates'
+        to:
+          $ref: '#/components/schemas/options_coordinates'
+    options_coordinates:
+      type: object
+      properties:
+        lat:
+          $ref: '#/components/schemas/lat'
+        lon:
+          $ref: '#/components/schemas/lon'
+    options_meta:
+      type: object
+      properties:
+        name:
+          type: string
+        description:
+          type: string
+        image:
+          type: string
+          format: url
+        car:
+          type: object
+          properties:
+            passengers:
+              type: integer
+    leg:
+      type: object
+      description: >-
+        A OpenTripPlanner compatible definition of a leg (see OpenTripPlanner
+        docs for reference)
```

Comment:
Would be good to align this with [Segment](https://github.com/maas-alliance/apis/blob/master/maas-api.yaml#L2436). Might make sense to make the Segment be based on OTP and adding the additional fields, though see comments on `time` and `mode` below.

Reply:
OTP specs: http://dev.opentripplanner.org/apidoc/1.0.0/json_Leg.html

Reply:
If we do base this on OTP, can you confirm that we can use this? It is licensed under [LGPL-3.0](https://choosealicense.com/licenses/lgpl-3.0/).

Reply:
Let's make this an enhancement issue, too.

Reply:
Enhancement https://github.com/maas-alliance/apis/issues/18